app/repositories/onlinetesting/Phase1TestRepository.scala

```diff
@@ -92,6 +92,22 @@ class Phase1TestMongoRepository(dateTime: DateTimeFactory)(implicit mongo: () =>
     selectRandom[OnlineTestApplication](query, maxBatchSize)
   }
 
+  def nextSdipFaststreamCandidateReadyForSdipProgression: Future[Option[Phase1TestGroupWithUserIds]] = {
+    val query = BSONDocument("$and" -> BSONArray(
+      BSONDocument("applicationRoute" -> ApplicationRoute.SdipFaststream),
+      BSONDocument("testGroups.PHASE1.evaluation.result" -> BSONDocument("$elemMatch" -> BSONDocument("scheme" -> SchemeType.Sdip)))
+    ))
+    collection.find(query).one[BSONDocument] map {
+      case Some(doc) =>
+        val applicationId = doc.getAs[String]("applicationId").get
+        val userId = doc.getAs[String]("userId").get
+        val bsonPhase1 = doc.getAs[BSONDocument]("testGroups").map(_.getAs[BSONDocument](phaseName).get)
+        val phase1TestGroup = bsonPhase1.map(Phase1TestProfile.bsonHandler.read).get
```

Comment:
the mongo query used means that the test profile sub-document is always present, so I thought it would be fine to use the get rather mapping an option and throwing an exception if it was not defined.